### DISCUSSION TITLE
Unraid "Extra Parameters" for Drives Monitoring

### DISCUSSION THREAD

**Discussion by Isanderthul** [guydavis/machinaris#641] [Q&A]
When I run `sudo smartctl -i /dev/sda` I get an error that the bridge type is unknown
However if I specify the bridge type as scsi then the smartctl works
How do I pass the `-d scsi` into the extra parameters as per https://github.com/guydavis/machinaris/wiki/Drives#unraid?

```
root@oliver:~# sudo smartctl -i -d scsi /dev/sda
smartctl 7.1 2019-12-30 r5022 [x86_64-linux-5.10.28-Unraid] (local build)
Copyright (C) 2002-19, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Vendor:               SanDisk
Product:              Ultra Fit
Revision:             1.00
Compliance:           SPC-4
User Capacity:        15,376,318,464 bytes [15.3 GB]
Logical block size:   512 bytes
Serial number:        0501b74080a57367a182
Device type:          disk
Local Time is:        Tue Apr  5 11:38:06 2022 SAST
SMART support is:     Available - device has SMART capability.
SMART support is:     Enabled
Temperature Warning:  Disabled or Not Supported
```

**Comment by guydavis** [comment]
Interesting!  Please paste screenshots showing:

1. Full contents of your "Extra Parameters" field in the Machinaris docker configuration of Unraid.
2. `smartctl --scan` output from inside the container.
3. `smartctl -a /dev/sda`
4. `smartctl -a -d scsi /dev/sda`

This will help me understand the situation as I have different hardware.  Thanks!

**Reply by Isanderthul** [reply]
> 1. Full contents of your "Extra Parameters" field in the Machinaris docker configuration of Unraid.

`-h $HOSTNAME -v /run/udev:/run/udev:ro --cap-add SYS_RAWIO --device=/dev/sdb:/dev/sdb --device=/dev/sdc:/dev/sdc --device=/dev/sdd:/dev/sdd --device=/dev/sde:/dev/sde --device=/dev/sdf:/dev/sdf`

> 2. `smartctl --scan` output from inside the container.

```
root@oliver:~# sudo smartctl --scan
/dev/sdb -d scsi # /dev/sdb, SCSI device
/dev/sdc -d scsi # /dev/sdc, SCSI device
/dev/sdd -d scsi # /dev/sdd, SCSI device
/dev/sde -d scsi # /dev/sde, SCSI device
/dev/sdf -d scsi # /dev/sdf, SCSI device
```

> 3. `smartctl -a /dev/sda` 

```
root@oliver:~# smartctl -a /dev/sda
smartctl 7.1 2019-12-30 r5022 [x86_64-linux-5.10.28-Unraid] (local build)
Copyright (C) 2002-19, Bruce Allen, Christian Franke, www.smartmontools.org

/dev/sda: Unknown USB bridge [0x0781:0x5583 (0x100)]
Please specify device type with the -d option.

Use smartctl -h to get a usage summary
```

> 4. `smartctl -a -d scsi /dev/sda`

```
root@oliver:~# smartctl -a -d scsi /dev/sda
smartctl 7.1 2019-12-30 r5022 [x86_64-linux-5.10.28-Unraid] (local build)
Copyright (C) 2002-19, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Vendor:               SanDisk
Product:              Ultra Fit
Revision:             1.00
Compliance:           SPC-4
User Capacity:        15,376,318,464 bytes [15.3 GB]
Logical block size:   512 bytes
Serial number:        0501b74080a57367a182
Device type:          disk
Local Time is:        Fri Apr  8 12:14:39 2022 SAST
SMART support is:     Available - device has SMART capability.
SMART support is:     Enabled
Temperature Warning:  Disabled or Not Supported

=== START OF READ SMART DATA SECTION ===
SMART Health Status: OK
Current Drive Temperature:     0 C
Drive Trip Temperature:        0 C

Error Counter logging not supported

Device does not support Self Test logging
```

**Reply by guydavis** [reply]
Thanks for the response.  You opened this ticket regarding `/dev/sda` however I don't see that in your "Extra Parameters" passed to Unraid.  Please either add that to your Extra Parameters or switch out `/dev/sda` in the commands I requested for a drive that is in your Extra Parameters such as `/dev/sdb`.    In particular, it should be listed by `smartcl --scan`.

**Comment by Isanderthul** [comment]
not sure if this is useful, but this project also made a collector config file https://github.com/AnalogJ/scrutiny/pull/88
